tools/ios-versioning/index.js

```diff
@@ -0,0 +1,183 @@
+'use strict';
+
+const chalk = require('chalk');
+
+const transformFunctions = [
+  // TODO: move previous transforming steps here as well
+  injectMacros,
+  postTransforms,
+];
+
+function injectMacros({ versionPrefix }) {
+  return [
+    {
+      // add a macro ABIXX_0_0EX_REMOVE_VERSION(str) to RCTDefines
+      paths: 'RCTDefines.h',
+      replace: /(.|\s)$/,
+      with: `$1\n#define ${versionPrefix}EX_REMOVE_VERSION(string) (([string hasPrefix:@"${versionPrefix}"]) ? [string stringByReplacingCharactersInRange:(NSRange){0,@"${versionPrefix}".length} withString:@""] : string)\n`,
+    },
+    {
+      // use the macro on the return value of `RCTBridgeModuleNameForClass`
+      // to pass unversioned native module names to JS
+      paths: 'RCTBridge.m',
+      replace: /(return ABI\d+_\d+_\d+RCTDropReactABI\d+_\d+_\d+Prefixes)\(name\)/g,
+      with: `$1(${versionPrefix}EX_REMOVE_VERSION(name))`,
+    },
+    {
+      // use the macro on the return value of `moduleNameForClass`
+      // to pass unversioned native module names to JS
+      paths: 'RCTComponentData.m',
+      replace: /(if \(\[name hasPrefix:@"RK"\]\) \{\n)/g,
+      with: `name = ${versionPrefix}EX_REMOVE_VERSION(name);\n  $1`,
+    },
+    {
+      // injects macro into `enqueueJSCall:method:args:completion:` method of RCTCxxBridge
+      paths: 'RCTCxxBridge.mm',
+      replace: /callJSFunction\(\[module UTF8String\],/,
+      with: `callJSFunction([${versionPrefix}EX_REMOVE_VERSION(module) UTF8String],`,
+    },
+    {
+      // now that this code is versioned, remove meaningless EX_UNVERSIONED declaration
+      paths: 'EXUnversioned.h',
+      replace: /(#define symbol[.\S\s]+?(?=\n\n)\n\n)/g,
+      with: '\n',
+    },
+  ]
+}
+
+function postTransforms({ versionPrefix }) {
+  return [
+    // react-native
+    {
+      paths: ['RCTRedBox.m', 'RCTLog.mm'],
+      replace: /#if (ABI\d+_\d+_\d+)RCT_DEBUG/g,
+      with: '#if $1RCT_DEV',
+    },
+    {
+      paths: ['NSTextStorage+FontScaling.h', 'NSTextStorage+FontScaling.m'],
+      replace: /NSTextStorage \(FontScaling\)/,
+      with: `NSTextStorage (${versionPrefix}FontScaling)`,
+    },
+    {
+      paths: ['NSTextStorage+FontScaling.h', 'NSTextStorage+FontScaling.m', 'RCTTextShadowView.m'],
+      replace: /\b(scaleFontSizeToFitSize|scaleFontSizeWithRatio|compareToSize)\b/g,
+      with: `${versionPrefix.toLowerCase()}_rct_$1`,
+    },
+    {
+      paths: 'RCTWebView.m',
+      replace: /@"ReactABI\d+_\d+_\d+-js-navigation"/,
+      with: '@"react-js-navigation"',
+    },
+
+    // Universal modules
+    {
+      paths: `UniversalModules/${versionPrefix}EXScoped`,
+      replace: /(EXScopedReactABI\d+_\d+_\d+Native)/g,
+      with: 'EXScopedReactNative',
+    },
+
+    // react-native-maps
+    {
+      paths: 'AIRMapWMSTile',
+      replace: /\b(TileOverlay)\b/g,
+      with: `${versionPrefix}$1`,
+    },
+    {
+      paths: 'AIRGoogleMapWMSTile',
+      replace: /\b(WMSTileOverlay)\b/g,
```

Comment:
Maybe instead of posttransform here, we should extend https://github.com/expo/expo/blob/c2487520174dfda398636c2e21a70e4c22bb5a24/tools/ios-tasks.js#L96 ?

Reply:
Imho extending such long and complicated patterns doesn't make much sense, I guess it's already pretty slow pattern and applying another case here would complicate and slow it down even more. Would it really make sense to apply this on all files? 
Also, I'd say these `WMSTileOverlay` are like edge cases, we have a set of prefixes (e.g. `EX`, `RCT`, `UM`) that we definitely want to version and search for them in the whole codebase, but others could probably be applied only on the subset of files. For example, do we use `AIR`-prefixed refs outside of the Maps/GoogleMaps folders? I don't think so, so this prefix could be versioned only in these folders, right?